pb_decode.c

```diff
@@ -294,9 +327,23 @@ bool checkreturn pb_decode_tag(pb_istream_t *stream, pb_wire_type_t *wire_type,
     *eof = false;
     *wire_type = (pb_wire_type_t) 0;
     *tag = 0;
-    
-    if (!pb_decode_varint32_eof(stream, &temp, eof))
+
+    if (stream->bytes_left == 0)
     {
+        *eof = true;
+        return false;
+    }
+
+    if (!pb_decode_varint32(stream, &temp))
+    {
+#ifndef PB_BUFFER_ONLY
+        /* band-aid for issue #1017. eof has been hit, but may have a lingering io error due to callbacks */
+        if (stream->callback != buf_read && stream->bytes_left == 0)
+        {
+            PB_SET_ERROR(stream, NULL);
```

Comment:
I think this is no-op because `PB_SET_ERROR` is:

     #define PB_SET_ERROR(stream, msg) (stream->errmsg = (stream)->errmsg ? (stream)->errmsg : (msg))

I think it is unnecessary, as the prior implementation didn't null out any error message either.

There is a subtle difference between this and the previous one: the previous one would only allow valid EOF at the first byte of the tag. But there is no guarantee about detecting corrupted messages anyway, so this is probably fine.

Reply:
Ah, right, I didn't actually test the functionality of canceling out the error. I just noticed that possibility while trying to track down issues that came up from my main loop not having a condition on bytes_left like the one in pb_decode.c